src/neo/Ledger/MemoryPool.cs

```diff
@@ -333,12 +369,13 @@ internal bool TryRemoveUnVerified(UInt256 hash, out PoolItem item)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal void InvalidateVerifiedTransactions()
+        internal void InvalidateVerifiedTransactions(StoreView snapshot)
         {
             foreach (PoolItem item in _sortedTransactions)
             {
                 if (_unverifiedTransactions.TryAdd(item.Tx.Hash, item))
                     _unverifiedSortedTransactions.Add(item);
+                item.CheckOracleResponse(snapshot);
```

Comment:
Why check it when setting the transactions as unverified?

Reply:
You are right, Please check https://github.com/neo-project/neo/pull/1641